### PR TITLE
Fixes notifications on new comment

### PR DESCRIPTION
**Pull request by naps62**
The previous implementation relied on `touch: true` to automatically save the
discussion when a new comment was created. Aparently, this does not trigger
active record callbacks, so subscriptions where not being updated.

Instead of adding yet another active_record callback, and starting to get too
much spaghetti there, I decided to move this to the controller action. I also
added a test for this

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def show
 
   def create
     comment = discussion.comments.create(comment_params)
+    discussion.update_subscriptions
```
Comment (zamith) [inline app/controllers/comments_controller.rb]: I don't think it's the controller's job to know this. I propose that we create an object to handle this. Something like a `CommentCreator`, or `CreatesComment`, or even `Commenter`. 

This make the job of the controller much clearer and to the point. Get HTTP request to create comment, parse params, tell someone to create a comment with the params, respond accordingly.

File: app/models/discussion.rb [hunk 2]
```diff
@@ -8,10 +8,6 @@ class Discussion < ActiveRecord::Base
 
   default_scope -> { order('updated_at DESC') }
 
-  after_save :update_subscriptions
```
Comment (zamith) [inline app/models/discussion.rb]: Have you tried an `after_commit` callback here? It seems that touch will trigger it. 

There is also `after_touch` which would make more sense, but it apparently has [this issue](https://github.com/rails/rails/issues/8759). We can give it a go, anyway.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (naps62) [inline app/controllers/comments_controller.rb]: Ok, where do we store this class? `app/services`?

File: app/models/discussion.rb (hunk 2 above)
Comment (naps62) [inline app/models/discussion.rb]: Well, after_touch seems to work. I can leave it like that for now then, but I believe we'll eventually have to change to solutions like `CommentCreator` if this notifications system gets more complex.
But for now it seems to be enough

Comment (zamith) [inline app/models/discussion.rb]: But does it get triggered more than once? That was the issue...

Comment (naps62) [inline app/models/discussion.rb]: I tested that. doesn't seem to be happening. But I guess I can add a test for that, just to be sure

Comment (naps62) [inline app/models/discussion.rb]: Actually there's no problem here. Even if the callback gets triggered twice in a transaction, the subscription won't change states again, and no new notification will be generated

Comment (zamith) [inline app/models/discussion.rb]: From what I could see, this meant two calls to the db. One that saves it, and an update. I don't see the need for that.

Comment (naps62) [inline app/models/discussion.rb]: It's one to save the comment, and another to touch the discussion. It's unnecessary, but it's also not problematic. And what would be the alternative?

Comment (zamith) [inline app/models/discussion.rb]: Sorry. I was reading in the email I mixed up the prs. It's ok with me. If we see this getting to slow, we can act on it later.

**Comment by zamith** [comment]
:+1:

**Pull request closed without merging**